docs/service-fabric/refactor-migrated-app.md

```diff
@@ -0,0 +1,155 @@
+
+Note that operations to persist dequeued items from a ReliableConcurrentQueue should ideally be idempotent. If an exception is thrown during the processing of an item from the queue, the same item may be processed more than once. In the Surveys application, the operation to merge survey answers to the *Tailspin.SurveyAnalysisService* is not idempotent because Tailspin decided that the survey analysis data is only a current snapshot of the analysis data and does not need to be consistent. The survey answers saved to Azure Blob Storage are eventually consistent, so the survey final analysis can always be recalculated correctly from this data.
+
+## Communication framework
+
+Each service in the Surveys application communicates using a RESTful web API. RESTful APIs offer the following benefits:
+* Ease of use: each service is built using ASP.Net Core MVC, which natively supports the creation of Web APIs.
+* Security: While each service does not require SSL, Tailspin could require each service to do so. 
+* Versioning: clients can be written and tested against a specific version of a web API.
+
+Services in the Survey application make use of the [reverse proxy][reverse-proxy] implemented by Service Fabric. Reverse proxy is a service that runs on each node in the Service Fabric cluster and provides endpoint resolution, automatic retry, and handles other types of connection failures. To use the reverse proxy, each RESTful API call to a specific service is made using a predefined reverse proxy port.  For example, if the reverse proxy port has been set to **19081**, a call to the *Tailspin.SurveyAnswerService* can be made as follows:
+
+```csharp
+static SurveyAnswerService()
+{
+    httpClient = new HttpClient
+    {
+        BaseAddress = new Uri("http://localhost:19081/Tailspin/SurveyAnswerService/")
+    };
+}
+```
+To enable reverse proxy, specify a reverse proxy port during creation of the Service Fabric cluster. For more information, see [reverse proxy][reverse-proxy] in Azure Service Fabric.
+
+## Performance considerations
+
+Tailspin created the ASP.NET Core services for *Tailspin.Web* and *Tailspin.Web.Surveys.Public* using Visual Studio templates. By default, these templates include logging to the console. Logging to the console may be done during development and debugging, but all logging to the console should be removed when the application is deployed to production.
+
+> [!NOTE]
+> For more information about setting up monitoring and diagnostics for Service Fabric applications running in production, see [monitoring and diagnostics][monitoring-diagnostics] for Azure Service Fabric.
+
+For example, the following lines in *startup.cs* for each of the web front end services should be commented out:
+
```

Comment:
Maybe say something about replacing "AddConsole" with another logging sink? That way we're not encouraging customers to get rid of logging altogether, and it doesn't look like a bug in the VS template.

Reply:
Could point to this doc? https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?tabs=aspnetcore2x#built-in-logging-providers

Reply:
We found that the Console logging sink performed poorly and not needed in production. The referenced monitoring and diagnostics article talks deeply about how to properly log. Do we need to implement here?

Reply:
I don't think we need to add anything to the RI - but in the doc, call out that console logging sink should be replaced by a different (production performance) logging sink.

Reply:
I agree.

Reply:
Do we want to be discussing logging in the performance considerations section? There's currently a link to the SF monitoring and diagnostics  document.